clang/lib/CodeGen/CodeGenTypes.cpp

```diff
@@ -104,7 +104,10 @@ llvm::Type *CodeGenTypes::ConvertTypeForMem(QualType T) {
   if (T->isConstantMatrixType()) {
     const Type *Ty = Context.getCanonicalType(T).getTypePtr();
     const ConstantMatrixType *MT = cast<ConstantMatrixType>(Ty);
-    return llvm::ArrayType::get(ConvertType(MT->getElementType()),
+    llvm::Type *IRElemTy = ConvertType(MT->getElementType());
+    if (T->isConstantMatrixBoolType() && Context.getLangOpts().HLSL)
```

Comment:
```suggestion
    if (Context.getLangOpts().HLSL && T->isConstantMatrixBoolType())
```
Just a nit - I would flip the conditions. `getLangOpts` is likely faster than `isConstantMatrixBoolType` so in non-HLSL cases it will exit earlier.